wsmaster/codenvy-hosted-sso-auth-bearer/src/test/java/com/codenvy/auth/sso/server/BearerTokenAuthenticationServiceTest.java

```diff
@@ -77,4 +90,20 @@ public void shouldSendEmailToValidateUserEmailAndUserName() throws Exception {
         assertEquals(argumentCaptorValue.getFrom(), "noreply@host");
         assertEquals(argumentCaptorValue.getSubject(), "Verify Your Codenvy Account");
     }
+
+    @Test
+    public void shouldThrowAnExceptionWhenUserBeyondTheLicense() throws Exception {
+        bearerTokenAuthenticationService.mailSender = "noreply@host";
+        ValidationData validationData = new ValidationData("Email", "UserName");
+        when(licenseManager.canUserBeAdded()).thenReturn(false);
+
+        Response response = given().contentType(ContentType.JSON).content(validationData).post("/internal/token/validate");
+
+        assertEquals(response.getStatusCode(), 403);
+        assertEquals(response.prettyPrint(), String.format("{\n"
```

Comment:
I'd parse response body to ServiceError with DtoFactory and check message instead of json